source/slang/slang-emit-spirv.cpp

```diff
@@ -1213,6 +1223,19 @@ struct SPIRVEmitContext
         return m_NonSemanticDebugPrintfExtInst;
     }
 
+    SpvStorageClass addressSpaceToStorageClass(AddressSpace addrSpace)
+    {
+        switch (addrSpace)
+        {
+        case AddressSpace::Generic:
+            return SpvStorageClassMax;
```

Comment:
I have a question here, in the PR description, I see
"and a variable will have a pointer type with the Generic address space",

but SpvStorageClassMax is an invalid storage class, right?

Reply:
If the address space is `Uniform`, which is 4 in your definition.

so it will return `SpvStorageClassWorkgroup`.

Shouldn't return `SpvStorageClassUniform`?

Reply:
`Generic` address space means there is no specific address space info associated with the pointer type. It is the responsibility of the spirv legalization and address propagation step to make sure all pointers in generic address space are turned into pointers with a valid SPIRV Storage class as its address space.